generated/Pet.h

```diff
@@ -0,0 +1,68 @@
+#ifndef ORG_OPENAPITOOLS_MODEL_Pet_H_
+#define ORG_OPENAPITOOLS_MODEL_Pet_H_
+
+#include "serialization.h"
+
+#include <optional>
+
+#include "Tag.h"
+#include <string>
+#include "Category.h"
+#include <vector>
+
+namespace org {
+namespace openapitools {
+namespace model {
+
+/**
+ * \brief A pet for sale in the pet store
+ */
+struct Pet {
+    static const char *dtoName() { return "swagger::dto::Pet"; }
+
+    // enums
```

Comment:
Yay, enums! Caveats:
 [ ] It's currently broken, as we need to generate from/to string
 [ ]  `Enum` is appended to the name. Comes from the field `enumName` in the generated model. This is needed so the enum isn't named the same as the field that uses it. Not sure if there's really another way.

Reply:
Yes, please

Reply:
`from_json` and `to_json` serializers are now generated for enums. Here's an example: https://github.com/hrantzsch/openapi-codegen-test/pull/1/files#diff-cadf6a5e904563e0dc657030f21d6993R68

Reply:
> Enum is appended to the name. Comes from the field enumName in the generated model. This is needed so the enum isn't named the same as the field that uses it. Not sure if there's really another way.

Really? But this enum could be called `Status` and can totally be the type of the `status` member, simply because of casing, no?

Or do you mean, that we would be at the mercy of the underlying swagger definition? I.e. would it be possible to define both a type and a data member with the exact same name?
In that case, I'd suggest to add a scope for all defined enums like so:

```C++
struct Pet {
  struct Enums {
    enum class status {
      AVAILABLE,
      PENDING,
      SOLD
    };
  };

  Enums::status status;
};
```

Its disputable whether that would be nicer, of course.

Reply:
Hm, the `struct Enums` version looks a bit awkward to me.

Calling the enum `Status` would be the optimal solution of course. But I haven't found a way to access something like `{{nameInUpperCase}}` from the template side, yet. So I'm using what the generator provides me as `enumName`.
Looking into the generator code a bit more, I might be able to customize this, though. I'll try.

Reply:
It's fancy now :)